### PR TITLE
Create DesignDocumentation.md

### PR DESCRIPTION
**Pull request by preetkaran20**
Draft PR. Working on adding this information.

### CODE REVIEW

**Comment by preetkaran20** [comment]
@hemantgs i am writing this documentation. please add your inputs in case i have missed something.

File: docs/DesignDocumentation.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Technical Design Documentation
+
+While designing VulnerableApp, major emphasis was given on **Ease of adding Vulnerabilities** such that developers of Vulnerability Scanners need to put minimal effort for adding new Vulnerabilities for testing there payloads/attack vectors.
```
Comment (hemantgs) [inline docs/DesignDocumentation.md:3]: adding new Vulnerabilities for testing their payload/

File: docs/DesignDocumentation.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+## Design:
+UserInterface for VulnerableApp is driven from backend endpoint "/allEndPointJson" which provides entire information about all the Vulnerabilities present in the VulnerableApp. This information is dynamic and is generated from the annotations present over the java class.
+
+### Annotations:
+There are 3 annotations which are driving entire VunerableApp, specifically UserInterface, Vulnerability definition for **Scanners**, Exposing Vulnerabilities and its help etc.
```
Comment (hemantgs) [inline docs/DesignDocumentation.md:9]: typo : VunerableApp

File: docs/DesignDocumentation.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+All the other parameters in the annotations are used for generating information for scanners, specifically */scanner* endpoint. 
+1. [Java Documentation](https://github.com/SasanLabs/VulnerableApp/blob/master/src/main/java/org/sasanlabs/internal/utility/annotations/VulnerableAppRequestMapping.java)
+2. [Example usage](https://github.com/SasanLabs/VulnerableApp/blob/master/src/main/java/org/sasanlabs/service/vulnerability/pathTraversal/PathTraversalVulnerability.java)
+3. [Example Templates](https://github.com/SasanLabs/VulnerableApp/tree/master/src/main/resources/static/templates)
+
+**AttackVector**:
+This annotation is mainly used to help user's to expose the vulnerability. This annotation is used to help students/security enthusiasts about various payloads/ways to expose the vulnerability.
```
Comment (hemantgs) [inline docs/DesignDocumentation.md:39]: typo : users

**Review by hemantgs — APPROVED** [review]
Just a few nits
Other than that cool !

**Comment by preetkaran20** [comment]
@hemantgs i am done with the final changes. please review.

**Comment by preetkaran20** [comment]
Merging for now. Will address review comments in next PR.

**Pull request merged**